### DISCUSSION TITLE
Add option for REPL timeout

### DISCUSSION THREAD

**Discussion by gaunap** [webdriverio/webdriverio#8025] [Q&A]
I use WDIO-REPL for playing around with wdio commands on iOS. Alas this is a little cumbersome, since the connection to Appium breaks up every 60s if no new command is send. I see no way to configure the timeout (_newCommandTimeout_ I guess).
My setup is a started iphone simulator, a running Appium server and wdio.
I start wdio using `wdio repl ios --platformVersion 14.3 --deviceName 'iPhone 11 Pro Max'`

I see no command line arg which can configure a timeout. Am I missing smth ?

**Comment by christian-bromann** [comment]
Can you try with:

```sh
wdio repl ios --platformVersion 14.3 --deviceName 'iPhone 11 Pro Max' --connectionRetryTimeout 9999999
```

**Reply by gaunap** [reply]
This doesn't help. I still get the following result:
```
[BaseDriver] Shutting down because we waited 60 seconds for a command
[Appium] Closing session, cause was 'New Command Timeout of 60 seconds expired. Try customizing the timeout using the 'newCommandTimeout' desired capability'
```
As I understand the wdio repl command generates a capability and passes it to appium. _connectionRetryTimeout_ doesn't add this value to the capabilities object.

**Reply by christian-bromann** [reply]
This is a timeout that can be configured on the [Appium side](https://appium.io/docs/en/writing-running-appium/caps/) (see `newCommandTimeout`). You could also pass it in as capability but I am not sure if that works via the repl command, maybe try:

```sh
wdio repl ios --platformVersion 14.3 --deviceName 'iPhone 11 Pro Max' --"appium:newCommandTimeout" 9999999
```

**Reply by gaunap** [reply]
This argument doesn't work.
And I don't see a way to configure it on Appium side using the Appium-Server-GUI.

**Reply by christian-bromann** [reply]
Please create an issue then to allow more capabilities to be passed in to the repl command.

**Reply by gaunap** [reply]
done: https://github.com/webdriverio/webdriverio/issues/8048
